hocjordan/handeye_calib_camodocal
# Refactor

// Every PR needs 2 reviews: the team is automatically added, but please ask
// for a second review from a relevant member.

## Summary
Allows calibrating the camera using a ros service, as we do with VISP.

## Overview
// What does this change do? What is its purpose?

## Implementation details
Camodocal as implemented here: https://github.com/jhu-lcsr/handeye_calib_camodocal requires the user to manually input commands (save data point, delete previous data point, quit). A fork of this repo implemented a way to input these commands using a ros service (input string) in `handeye_calibration_auto.cpp`. I refactored this script to get rid of code duplication and to rely more closely on the original code `handeye_calibration.cpp`. I also removed a weird block of code that was overwriting the calibration and saving the marker_bundle > world transform.

## Risks
// Is there any chance of this PR breaking anything? Was there an API change?
// Anything that needs to be handled by users of this code?

## How to test?
// Is there a way to check that the behaviour is correct?
// Please mention any tests that have been run (including on the real robot) or new tests that have been added to the repo.
